src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs

```diff
@@ -155,7 +155,7 @@ private static IAsyncStateMachineBox GetStateMachineBox<TStateMachine>(
             [NotNull] ref Task<TResult>? taskField)
             where TStateMachine : IAsyncStateMachine
         {
-            ExecutionContext? currentContext = ExecutionContext.Capture();
+            ExecutionContext? currentContext = ExecutionContext.CaptureForSuspension(Thread.CurrentThread);
```

Comment:
What about the non-T version of this?

Also cc @stephentoub

Reply:
non-T versions internally delegate to `Task<VoidTaskResult>`, same for `ValueTask` and `ValueTask<T>`